File: src/main/java/com/example/speedyprintserver/ENTITY/Service/Impression.java
package com.example.speedyprintserver.ENTITY.Service;

import com.example.speedyprintserver.ENTITY.Commande.Face;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table
@Data
public class Impression {
    //Prix d'mpression avec quantite
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idImpression ;

    //l'attribut quantite a été remplacé par borneInferieure et borneSuperieure

    private Integer borneInferieure;
    private Integer borneSuperieure;
    private Double remise ;
    private Double montant ;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Service service ;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private ModeImpression modeImpression;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Dimension dimension;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Format format;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private TypePapier typePapier;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Faces faces;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Grammage grammage;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Finition finition;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Forme forme;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Couleur couleur;
}

File: speedyprintapp-webapp/src/app/core/Model/Parametre/Impression.ts
import { Conception } from "./Conception";
import { Couleur } from "./Couleur";
import { Dimension } from "./Dimension";
import { Faces } from "./Faces";
import { Finition } from "./Finition";
import { Format } from "./Format";
import { Forme } from "./Formes";
import { Grammage } from "./Grammage";
import { ModeImpression } from "./ModeImpression";
import { Service } from "./Service";
import { TypePapier } from "./TypePapier";

export class Impression {
  idImpression ?:number ;
  borneInferieure !:number ;
  borneSuperieure !:number ;
  remise!:number;
  montant!:number;
  service !:Service;

  modeImpression !:ModeImpression;
  dimension!:Dimension;
  format!:Format;
  conception!:Conception;
  typePapier ?:TypePapier;
  face ?: Faces;
  couleur ?:Couleur;
  grammage ? : Grammage;
  finition ?:Finition ;
  forme ? : Forme ;
}
